cmd/util/ledger/migrations/cadence_value_validation.go

```diff
@@ -0,0 +1,599 @@
+package migrations
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/onflow/atree"
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/rs/zerolog"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/util"
+	"github.com/onflow/flow-go/ledger"
+)
+
+var nopMemoryGauge = util.NopMemoryGauge{}
+
+// TODO: optimize memory by reusing payloads snapshot created for migration
+func validateCadenceValues(
```

Comment:
Where is this function used?

Reply:
`validateCadenceValues()` is used in `AtreeRegisterMigrator.MigrateAccount()`.  Validation is enabled by `-validate` flag.

https://github.com/onflow/flow-go/blob/759ba39ad83b50ab144c794381ca89e6abf54c94/cmd/util/ledger/migrations/atree_register_migration.go#L118-L123